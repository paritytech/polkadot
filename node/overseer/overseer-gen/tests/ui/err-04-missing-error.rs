#![allow(dead_code)]

use polkadot_overseer_gen::*;

#[derive(Default)]
struct AwesomeSubSys;

#[derive(Clone, Debug)]
struct SigSigSig;

struct Event;

#[derive(Clone)]
struct MsgStrukt(u8);

#[overlord(signal=SigSigSig, event=Event, gen=AllMessages)]
struct Overseer {
	#[subsystem(MsgStrukt)]
	sub0: AwesomeSubSys,

	i_like_pie: f64,
}

#[derive(Debug, Clone)]
struct DummySpawner;

struct DummyCtx;

fn main() {
	let _ = Overseer::builder()
		.sub0(AwesomeSubSys::default())
		.i_like_pie(std::f64::consts::PI)
		.spawner(DummySpawner)
		.build()
		.unwrap();
}
