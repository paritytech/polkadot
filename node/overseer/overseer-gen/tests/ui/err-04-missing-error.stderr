error: Must declare the overseer error type via `error=..`.
  --> $DIR/err-04-missing-error.rs:16:1
   |
16 | #[overlord(signal=SigSigSig, event=Event, gen=AllMessages)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the attribute macro `overlord` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0433]: failed to resolve: use of undeclared type `Overseer`
  --> $DIR/err-04-missing-error.rs:30:10
   |
30 |     let _ = Overseer::builder()
   |             ^^^^^^^^ use of undeclared type `Overseer`
